### PR TITLE
Add single-tile index: GeoJSONVTTile

### PR DESCRIPTION
**Pull request by asheemmamoowala**
For _Custom Sources_ in GL-Native, a `GeoJSONVT` index is used to convert, clip, and tile `geojson` geometry from the client. For each high-zoom tile, this requires splitting the features for each tile, even though the features are already provided for only a specific tile [ref](https://github.com/mapbox/mapbox-gl-native/pull/9983/commits/b5e09a88b2e8f86e00709941b611dd5f863014ca#diff-5363356a8978f02458583cea92498957R47-48)

For a  simple LineString feature, fetching just a single tile 100 times:
GeoJSONVT : 498.2ms (avg: 5ms)
GeoJSONVTTile: 148.6ms (avg: 1ms)

[gl-native#9983](https://github.com/mapbox/mapbox-gl-native/pull/9983)

### CODE REVIEW

**Comment by asheemmamoowala** [comment]
> If we only need a single converted tile and not any of the GeoJSON-VT features like indexing and tiling, why expose the conversion through this interface?

This is a question I went back and forth on. I see this module as the logical place for conversion from GeoJSON data to Vector Tile data - even though it doesn't use a tile index or some of the other features.

> Why do we need to world-wrap features?

I left this in case tiled features are provided in a wrapped world where features cross the anti-meridian. It may not be necessary specifically for this API, but the data may already be in that format.

**Comment by mourner** [comment]
Yeah, I think it makes sense to do this in GeoJSON-VT. I'm just not sure whether it should really clip/ wrap tiles — having the data tiled should already imply clipping/wrapping, so doing this the second time after the conversion feels redundant, unless we want to do it as a "fixing" stage for badly tiled data.

**Comment by asheemmamoowala** [comment]
Would it be better if the clipping and wrapping steps were optional instead? It would allow someone to bring data that needs additional fixing and give them an efficient tool to do it, instead of requiring that the data be corrected at the platform level.

File: include/mapbox/geojsonvt.hpp [hunk 1]
```diff
@@ -60,6 +60,48 @@ inline uint64_t toID(uint8_t z, uint32_t x, uint32_t y) {
...
+        internalTile.emplace(
+            detail::InternalTile{ clippedFeatures, z, x, y, options.extent, options.buffer, tolerance });
+    }
+
+    const Tile& getTile() {
+        return internalTile->tile;
+    }
+
+private:
+    std::experimental::optional<detail::InternalTile> internalTile;
```
Comment (mourner) [inline include/mapbox/geojsonvt.hpp]: One more thing, does it have to be `optional`? I see that it's populated in constructor and then never removed dynamically, so perhaps we could do without `optional`?

**Comment by asheemmamoowala** [comment]
Thinking about this a bit more, I feel that the `GeoJSONVTTile` class is ill-suited here. All it does is compute the `Tile` in the constructor and return it in `getTile()`. It has no other state or operations. 

A static method makes more sense in this case, but I'm not sure if it belongs to the namespace or in `GeoJSONVT`.
```cpp
//static Namespace method
mapbox::geojson::geoJSONToTile(geojson, z, x, y, options, bool wrap, bool clip);

// static Member function
mapbox::geojson::GeoJSONVT::getTile(geojson, z, x, y, options, bool, wrap, bool clip);
```

@mourner - do you have any advice on which of these is more appropriate?

File: include/mapbox/geojsonvt.hpp (hunk 1 above)
Comment (asheemmamoowala) [reply include/mapbox/geojsonvt.hpp]: Doesn't need to be `optional`. In fixing this, I realized that this class doesn't need to maintain state and is better used as a method that returns a `Tile`

**Pull request merged**